### PR TITLE
Removing pipenv and replacing it with poetry

### PR DESCRIPTION
**Pull request by robons**
This PR (almost incidentally) upgrades the version of csvcubed to v0.1.3

### CODE REVIEW

**Review by mikeAdamss — APPROVED** [review]
This is tricky. I think you know as well as I do that there's quite a lot of (I'll admit clever) hackery in here, if nothing else we've got pyenv, poetry and naked venv manipulation, if feels like 2 out of 3 of those should be unnecessary.

On the flip side, we're basically code freezing gssutils, so "make it quicker until we turn it off" is fairly compelling.

So I'm gonna approve with one caveat on the approval - can you have a think about whether we really need to base this on `pythonversiontesting` please. That's going to be more big images on a Jenkins server already struggling with big images, it does concern me slightly.

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,49 @@
...
+
+# Write all dependencies to file.
+RUN poetry export --format requirements.txt --output /requirements.txt --without-hashes --dev
+
+# Install all dependencies listed in text file to the venv.
+RUN ${VENV_PIP} install --requirement /requirements.txt
+RUN ${VENV_PIP} install poetry
+
+# Install mkdocs for external docs building and publishing
+RUN python3 -m pip install mkdocs mkdocs-material mkdocs-mermaid2-plugin
```
[the remark this answers is not in the document]
Comment (mikeAdamss) [reply Dockerfile]: nvm, started review before you pulled this out

File: Dockerfile [hunk 2]
```diff
@@ -1,11 +1,49 @@
-FROM python:3.9
+FROM gsscogs/pythonversiontesting:v1.0.1
```
[the remark this answers is not in the document]
Comment (robons) [reply Dockerfile:1]: It brings poetry, docker (for testing) as well as pyright already installed. Also, it already exists as a pre-built image so gives us a build performance improvement on Jenkins.

**Comment by robons** [comment]
> This is tricky. I think you know as well as I do that there's quite a lot of (I'll admit clever) hackery in here, if nothing else we've got pyenv, poetry and naked venv manipulation, if feels like 2 out of 3 of those should be unnecessary.
> 
> On the flip side, we're basically code freezing gssutils, so "make it quicker until we turn it off" is fairly compelling.
> 
> So I'm gonna approve with one caveat on the approval - can you have a think about whether we really need to base this on `pythonversiontesting` please. That's going to be more big images on a Jenkins server already struggling with big images, it does concern me slightly.

I do agree it's a bit overkill, but we're already using this image elsewhere for all of the csvcubed builds - so it's already going to be cached on the server (once for all of them). So I believe it saves space over having separate docker builds for each of the projects.

**Pull request merged**